### ISSUE TITLE
Allow Faces.sendFile() to take some sort of OutputStreamCallback

### ISSUE THREAD

**Issue by kalgon** [omnifaces/omnifaces#194]
It would be nice if there was a `Faces.sendFile()` method which would take some sort of `OutputStreamCallback` (or a `Callback.WithArgument<OutputStream>`) like this:

    interface OutputStreamCallback {
        void writeInto(OutputStream outputStream) throws (IO)Exception;
    }

I am asking for this because some processes are naturally easier to use with `OutputStream` rather than `InputStream` (like marshalling a JAXB object).

    Faces.sendFile(os -> marshaller.marshal(myobject, os), "myfile.xml", true);

**Comment by BalusC** [comment]
idea is nice, but exception handling is irksome. I'm wary of declaring "throws Exception" in the API.

**Comment by kalgon** [comment]
I used a lambda here for brevity but I don't mind using an anonymous class that only throws IOException if that is OK for you.

As this method would not be in charge of writing on the output stream anymore, the (optional) content-length will have to be calculated by wrapping the response output stream in some sort of CountingByteOutputStream.

I came across the need for such a method one more time as I was generating PDFs with FOP.
